eldritch/test_mythos.py

```diff
@@ -2055,6 +2116,374 @@ def testNormal(self):
+    self.assertEqual(self.char.sanity, 3)
+
+  def testCanGainAtAsylum(self):
+    self.char.dollars = 2
+    self.char.place = self.state.places["Asylum"]
+    self.advance_turn(0, "encounter")
+    choice = self.resolve_to_choice(CardSpendChoice)
+    choice.resolve(self.state, "Restore 1 Sanity")
+    self.resolve_until_done()
+    self.assertEqual(self.char.sanity, 3)
+    self.advance_turn(1, "encounter")
+    choice = self.resolve_to_choice(CardSpendChoice)
+    self.assertEqual(choice.choices, ["Downtown Card", "Restore 1 Sanity", "Restore All Sanity"])
+    self.spend("dollars", 2, choice)
+    choice = self.resolve_to_choice(CardSpendChoice)
+    choice.resolve(self.state, "Restore All Sanity")
+    self.resolve_until_done()
+    self.assertEqual(self.char.sanity, 5)
+
+  def testCantGainFromOthers(self):
+    # TODO: Healing stone maybe shouldn't even be usable?
+    self.char.possessions.append(items.HealingStone(0))
+    self.advance_turn(0, "upkeep")
+    self.state.event_stack.append(UpkeepActions(self.char))
+    stone = self.resolve_to_usable(0, "Healing Stone0")
+    self.state.event_stack.append(stone)
+    self.resolve_until_done()
+    self.assertEqual(self.char.sanity, 2)
+    self.state.event_stack.append(Gain(self.char, {"sanity": 3}))
+    self.resolve_until_done()
+    self.assertEqual(self.char.sanity, 2)
+
+  def testCanGainOtherStuff(self):
+    self.state.event_stack.append(Gain(self.char, {"dollars": 1}))
+    self.resolve_until_done()
+    self.assertEqual(self.char.dollars, 1)
+
+
+class CloseLocationTest(EventTest):
+  def setUp(self):
+    super().setUp()
+    # Make sure there's enough monsters to draw twice
+    self.store = self.state.places["Store"]
+    self.shop = self.state.places["Shop"]
+    self.shoppe = self.state.places["Shoppe"]
+    self.char.place = self.store
+
+    matrix = monsters.FlameMatrix()
+    matrix.place = self.state.monster_cup
+    self.state.monsters.append(matrix)
+
+    self.mythos = Mythos10()
+    self.state.mythos.append(self.mythos)
+    self.advance_turn(0, "mythos")
+    self.resolve_until_done()
+    self.assertEqual(self.store.closed_until, 2)
+    self.assertEqual(self.shop.closed_until, 2)
+    self.assertEqual(self.shoppe.closed_until, 2)
+    self.assertIn(self.mythos, self.state.globals())
+
+  def tearDown(self):
+    self.advance_turn(2, "movement")
+    self.assertIsNone(self.store.closed_until)
+    self.assertIsNone(self.shop.closed_until)
+    self.assertIsNone(self.shoppe.closed_until)
+    self.assertNotIn(self.mythos, self.state.globals())
+    choice = self.resolve_to_choice(CityMovement)
+    self.assertIn("Store", choice.choices)
+    self.assertIn("Shop", choice.choices)
+    self.assertIn("Shoppe", choice.choices)
+
+  def testEvicted(self):
+    self.advance_turn(1, "movement")
+    self.assertEqual(self.char.place.name, "Rivertown")
+    choice = self.resolve_to_choice(CityMovement)
+    self.assertNotIn("Store", choice.choices)
```

Comment:
Does the character even have enough movement to make it to the shops?

Reply:
They do now that I've given them a Ruby!